### PR TITLE
HTBHF-1588 Add date of birth of children to the Claimant table and as…

### PR DESCRIPTION
**Pull request by mikeSmith08**
… an optional request parameter for a new claim.

Only validation on the request is that if there are any dates provided, they all need to be in the past.

### CODE REVIEW

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/ClaimantDTO.java [hunk 1]
```diff
@@ -75,4 +76,8 @@
     @ApiModelProperty(notes = "The claimant's email address. e.g. test@email.com")
     private String emailAddress;
 
+    @JsonProperty("childrenDob")
+    @ApiModelProperty(notes = "The dates of birth of the claimant's children (if they have any)")
+    private List<LocalDate> childrenDob;
```
Comment (dwybourn) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/ClaimantDTO.java]: Should have ListOfDatesInPast annotation.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/constraint/ListOfDatesInPastValidator.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    @Override
+    public boolean isValid(List<LocalDate> dates, ConstraintValidatorContext context) {
+        if (CollectionUtils.isEmpty(dates)) {
+            return true;
+        }
+        return allDatesInPast(dates);
+    }
+
+    private boolean allDatesInPast(List<LocalDate> dates) {
+        LocalDate now = LocalDate.now();
```
Comment (dwybourn) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/constraint/ListOfDatesInPastValidator.java]: For loop and below can be replaced with a one liner `return dates.stream().noneMatch(now::isBefore);`

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/entity/ClaimantTest.java [hunk 3]
```diff
@@ -298,4 +299,25 @@ void shouldValidateClaimantWithValidEmailAddress(String emailAddress) {
         //Then
         assertThat(violations).hasNoViolations();
     }
+
```
Comment (dwybourn) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/entity/ClaimantTest.java]: Can we have an empty list test as well.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/ClaimantDTO.java (hunk 1 above)
Comment (mikeSmith08) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/ClaimantDTO.java]: Yep, good spot. This does show that we don't have unit tests for the ClaimDTO, I'll add in another PR.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/constraint/ListOfDatesInPastValidator.java (hunk 2 above)
Comment (mikeSmith08) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/model/constraint/ListOfDatesInPastValidator.java]: I like the simplicity of my version, so I'm going to keep it, this one makes me think too much for something that is very simple.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/entity/ClaimantTest.java (hunk 3 above)
Comment (mikeSmith08) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/entity/ClaimantTest.java]: The first test does that by using `aValidClaimant()` which has no children's date of births in it.

**Pull request merged**